server/api/v1/project_service.go

```diff
@@ -93,7 +96,9 @@ func (s *ProjectService) UpdateProject(ctx context.Context, request *v1pb.Update
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	project, err := s.store.GetProjectV2(ctx, projectID)
```

Comment:
This should return the project too regardless of the deleted status.

I'm thinking of tackling GetEnvironmentV2 store to always to return the environment?

Reply:
But should we support update a removed resource?🤔

Reply:
Right, we should add a check similar to https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/api/v1/environment_service.go?L122.

And we should do the same check for Delete/Undelete() too. This will provide better error messages.

https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/api/v1/environment_service.go?L199

Reply:
Done